core/domain/classifier_services.py

```diff
@@ -127,6 +127,69 @@ def classify_string_classifier_rule(state, normalized_answer):
     return None
 
 
+
+
+def add_to_training_queue(exploration):
```

Comment:
@seanlip There are a few checks that a state needs to go through before it even comes to that. 
1. If a state is not a trainable state . i.e. if the state has no rule_specs/training-data, we don't even enter the 'if' condition. - Check the can_undergo_classification() function.
2. If a classifier does not already exist for a state, we don't check re-training.

So, when we break it down, only the few states that have training data actually even enter the main logic of this function. So, it actually calls is_retraining_necessary only for a very few select states. Do you still think this will be an issue?